File: CatalogueMovieSubmission2/app/src/main/java/com/azhara/cataloguemoviesubmission2/Adapter/TvShowAdapter.java
package com.azhara.cataloguemoviesubmission2.Adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.azhara.cataloguemoviesubmission2.Model.MovieTvShow;
import com.azhara.cataloguemoviesubmission2.R;
import com.bumptech.glide.Glide;

import java.util.ArrayList;

public class TvShowAdapter extends RecyclerView.Adapter<TvShowAdapter.TvShowHolder> {

    private ArrayList<MovieTvShow> listTv;
    private OnItemClickCallBack onItemClickCallBack;

    public TvShowAdapter(ArrayList<MovieTvShow> listTv){
        this.listTv = listTv;
    }

    public void setOnItemClickCallBack(OnItemClickCallBack onItemClickCallBack){
        this.onItemClickCallBack = onItemClickCallBack;
    }

    @NonNull
    @Override
    public TvShowHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_tvshow, parent, false);
        return new TvShowHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final TvShowHolder holder, int position) {
        MovieTvShow tvShow = listTv.get(position);
        Glide.with(holder.itemView.getContext())
                .load(tvShow.getPicture())
                .centerCrop()
                .into(holder.imgTv);
        holder.tvJudulTv.setText(tvShow.getJudul());
        holder.sinopisTv.setText(tvShow.getSinopsis());
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onItemClickCallBack.onItemClicked(listTv.get(holder.getAdapterPosition()));
            }
        });

    }

    @Override
    public int getItemCount() {
        return listTv.size();
    }

    public class TvShowHolder extends RecyclerView.ViewHolder {
        ImageView imgTv;
        TextView tvJudulTv, sinopisTv;
        public TvShowHolder(@NonNull View itemView) {
            super(itemView);

            imgTv = itemView.findViewById(R.id.img_tv);
            tvJudulTv = itemView.findViewById(R.id.tv_judul_tv);
            sinopisTv = itemView.findViewById(R.id.tv_sinopsis_tv);
        }
    }
    public interface OnItemClickCallBack{
        void onItemClicked(MovieTvShow tv);
    }
}

File: Submission3Movie/app/src/main/java/com/azhara/submission3movie/ui/tvshow/adapter/TVShowAdapter.kt
package com.azhara.submission3movie.ui.tvshow.adapter

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.azhara.submission3movie.R
import com.azhara.submission3movie.ui.tvshow.model.TVShow
import com.bumptech.glide.Glide
import kotlinx.android.synthetic.main.tvshow_item.view.*
import java.text.SimpleDateFormat

@Suppress("NULLABILITY_MISMATCH_BASED_ON_JAVA_ANNOTATIONS")
class TVShowAdapter : RecyclerView.Adapter<TVShowAdapter.TVShowHolder>() {

    private val listItems = ArrayList<TVShow>()
    private var onItemClickCallBack: OnItemClickCallBack? = null

    fun setData(list: ArrayList<TVShow>) {
        listItems.clear()
        listItems.addAll(list)
        notifyDataSetChanged()
    }

    fun setOnItemClickCallBack(onItemClickCallBack: OnItemClickCallBack) {
        this.onItemClickCallBack = onItemClickCallBack
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TVShowAdapter.TVShowHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.tvshow_item, parent, false)
        return TVShowHolder(view)
    }

    override fun getItemCount(): Int = listItems.size

    override fun onBindViewHolder(holder: TVShowAdapter.TVShowHolder, position: Int) {
        holder.bind(listItems[position])
    }

    inner class TVShowHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        @SuppressLint("SimpleDateFormat")
        fun bind(tvshow: TVShow) {
            //DateRelease ChangeFormat
            val date = tvshow.releaseDate
            val toSimpleDateFormat = SimpleDateFormat("yyyy-MM-dd")
            val parse = toSimpleDateFormat.parse(date)
            val changeFormat = SimpleDateFormat("d MMMM, yyyy")
            val releaseDate = changeFormat.format(parse)

            //Rating
            val getRate = tvshow.rating?.toFloat()
            val round = Math.round(getRate!!)
            val rating = round.toFloat() / 2

            //Picture path
            val pictureUrl = "https://image.tmdb.org/t/p/w342${tvshow.poster}"

            with(itemView) {
                tv_title_tvshow.text = tvshow.title
                rating_tvshow.rating = rating
                tv_release_tvshow.text = releaseDate
                Glide.with(itemView)
                    .load(pictureUrl)
                    .centerCrop()
                    .into(img_tvshow)

                itemView.setOnClickListener {
                    onItemClickCallBack?.onItemClicked(tvshow)
                }
            }
        }
    }

    interface OnItemClickCallBack {
        fun onItemClicked(data: TVShow)
    }

}